src/main/java/org/jenkinsci/plugins/oic/OicAlgorithmValidatorFIPS140.java

```diff
@@ -0,0 +1,129 @@
+package org.jenkinsci.plugins.oic;
+
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.impl.AESCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ContentCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ECDHCryptoProvider;
+import com.nimbusds.jose.crypto.impl.PasswordBasedCryptoProvider;
+import com.nimbusds.jose.crypto.impl.RSACryptoProvider;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import jenkins.security.FIPS140;
+
+/**
+ * This class helps in validating algorithms for FIPS-140 compliance and filtering the non-compliant algorithms when in
+ * FIPS mode.
+ */
+public class OicAlgorithmValidatorFIPS140 {
+
+    private static final boolean isFIPSMode = FIPS140.useCompliantAlgorithms();
+    private static final Set<JWSAlgorithm> jwsSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<JWEAlgorithm> jweSupportedAlgorithms = new LinkedHashSet<>();
+    private static final Set<EncryptionMethod> supportedEncryptionMethod = new LinkedHashSet<>();
+
+    // Below list of compliant algorithms will be used to block the FIPS non-compliant algorithms.
+    static {
+        // Init compliant Jws algorithms
+        jwsSupportedAlgorithms.addAll(MACSigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(RSASSASigner.SUPPORTED_ALGORITHMS);
+        jwsSupportedAlgorithms.addAll(ECDSASigner.SUPPORTED_ALGORITHMS);
+
+        // Init compliant Jwe algorithms
+        jweSupportedAlgorithms.addAll(AESCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(RSACryptoProvider.SUPPORTED_ALGORITHMS);
+        // RSA1_5 is deprecated and not a compliant algorithm.
+        jweSupportedAlgorithms.remove(JWEAlgorithm.RSA1_5);
+        jweSupportedAlgorithms.addAll(ECDHCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(PasswordBasedCryptoProvider.SUPPORTED_ALGORITHMS);
+
+        // Init complaint EncryptionMethods
+        supportedEncryptionMethod.addAll(ContentCryptoProvider.SUPPORTED_ENCRYPTION_METHODS);
+        // XC20P is not complaint method
+        supportedEncryptionMethod.remove(EncryptionMethod.XC20P);
+    }
+
+    /**
+     * Checks if the Jws signing algorithm used for OIC configuration is FIPS-140 compliant.
```

Comment:
although it doesn't.
as this returns `true` for something that is not compliant when not in FIPS mode.
Would it be better to just not perform the call to do any filtering if we are not in FIPS mode and then these methods would do what they are documented to do and what is expected by the function names?